### ISSUE TITLE
Is RN Swiper incompatible with React Navigation TabNavigator?

### ISSUE THREAD

**Issue by Q404** [leecade/react-native-swiper#389]
When I use Swiper inside one of TabNavigation screen, slide's content become invisible (dots and right-left buttons still visible).
![default](https://cloud.githubusercontent.com/assets/16742317/25135289/9f5e6dfa-245a-11e7-96fc-20c86a969531.PNG)

**Comment by qinfensky** [comment]
I have the same problem

**Comment by roman2710** [comment]
I used a swiper inside Tab Navigator and it works fine for me. Try disabling swiping in TabBarOptions.

**Comment by qinfensky** [comment]
@roman2710 swipeEnabled and scrollEnabled are two options were closed, but the problem still exists, the image cannot be displayed.

**Comment by roman2710** [comment]
Maybe there's something wrong with the picture. Try to render a simple <Text>Component

**Comment by Q404** [comment]
No, it's not. This component was tested in StackNavigator and it's working
fine...

**Comment by qinfensky** [comment]
Version:`"react": "~15.4.1",
    "react-native": "0.42.3",
    "react-native-swiper": "^1.5.4",`

No need for "StackNavigator", is a problem in "TabNavigator".

It works well in the ios emulator, but not in an Android emulator.

There are data, but no pictures are displayed.
<img width="360" alt="bug" src="https://cloud.githubusercontent.com/assets/7803520/25233441/66a6a424-2611-11e7-8ee0-033fc20ee3c6.png">

Code：
<img width="756" alt="code1" src="https://cloud.githubusercontent.com/assets/7803520/25233690/56f55646-2612-11e7-8bb2-85f4184d9c75.png">
<img width="771" alt="code2" src="https://cloud.githubusercontent.com/assets/7803520/25233696/5a991d3c-2612-11e7-8669-ea5436b121f7.png">

It is normal in the ios simulator:
<img width="375" alt="iosbug" src="https://cloud.githubusercontent.com/assets/7803520/25233891/f66e7afe-2612-11e7-8d33-a80cd02a5cc5.png">

tabBarOptions:
<img width="351" alt="tabbug" src="https://cloud.githubusercontent.com/assets/7803520/25234054/86a4a0da-2613-11e7-898d-a53bec33d655.png">


I have done a lot of tests, but I still don 't know the question, please help me to see what the problem is? thank you.

**Comment by jiasaichao** [comment]
@qinfensky  是TabNavigator的问题react-native-swiper安卓用的ViewPagerAndroid所以有的问题
看： https://github.com/react-community/react-navigation/issues/662

**Comment by qinfensky** [comment]
@jiasaichao 我找到问题所在了，是‘lazyLoad: true,’，设置为‘true’就可以显示了，但是还是有明显缺陷，就是‘initialRouteName: 'Home'’这个TabNavigator指定初始路由的react-native-swiper不显示，其余的三个都正常显示，这是什么问题呢？

**Comment by afinal** [comment]
@qinfensky 您好，我和你有同样的问题，首屏渲染的不可以显示，其他的都可以显示，请问这个问题你解决了吗？

**Comment by afinal** [comment]
我现在的临时解决方案是，默认跳转到其他的Tab，然后其他页面componentDidMount的时候，setTimeout跳转回到第一个Tab，首次进入的时候，我准备加个浮层进度条，要不然会看到跳转的过程，就尴尬了

**Comment by ayalpani** [comment]
The same happens to me. Anyone with a workaround?

**Comment by qinfensky** [comment]
@afinal @ayalpani 
这个问题我解决了，展示我的代码。
I solved the problem and show my code.
1.
<img width="800" alt="fixbug1" src="https://cloud.githubusercontent.com/assets/7803520/25464006/a6f35c82-2b2c-11e7-9cfa-fb3300f67173.png">
2.
<img width="360" alt="fixbug2" src="https://cloud.githubusercontent.com/assets/7803520/25464030/cab40bee-2b2c-11e7-822d-1293715385ef.png">

解决问题，大家玩得愉快
Solve the problem, everyone have a good time.

**Comment by jiasaichao** [comment]
@qinfensky @afinal 我不准备解决这个问题了，自己写一个TabBar，我对TabBar要求也不高，点击切换就完了，想怎么显示自己控制，只用react-navigation中的StackNavigator，这样好扩展而且也有可能不用react-navigation用react-native-navigation这样切换起来比较方便，我一直纠结是用react-navigation还是react-native-navigation。

**Comment by qinfensky** [comment]
@jiasaichao 我上面贴了代码，这样也能很好的显示了，navigation用起来确实方便。目前问题算是解决了~

**Comment by jiasaichao** [comment]
@qinfensky 嗯，解决了就行，估计以后bug解决了，你这个setTimeout也就不用加了。

**Comment by afinal** [comment]
@qinfensky 确实是个不错的方案，setTimeout 0也是可以的

**Comment by ayalpani** [comment]
Thank you @qinfensky 
Works!

**Comment by smartPiz** [comment]
still not working ...

**Comment by qinfensky** [comment]
@smartPiz show me the code

**Comment by StarOfLife** [comment]
如果我早点翻到的话，我就不发issues了

**Comment by qinfensky** [comment]
@StarOfLife 哈哈，现在 "react-native" 升级到了 "0.44.0"，"react-navigation"升级到了 "^1.0.0-beta.9"之后，不需要添加定时器了，直接渲染是没问题的，试试吧~

**Comment by StarOfLife** [comment]
@qinfensky 我现在就是用的这个版本的，然后出的这个问题。现在只能是先用着定时器了

**Comment by qinfensky** [comment]
@StarOfLife 嗯，OK，能解决问题就好，有问题随时联络~

**Comment by lehoi2195** [comment]
@qinfensky With version 0.44, it still does not work

**Comment by Adar-Li** [comment]
@qinfensky 您好，这个代码能私发我一下吗？

**Comment by gusgard** [comment]
What works for me was to set the `height` and wrap the `Swiper` in a `View` with `position absolute`, and also add `removeClippedSubviews={false}`

```
import { Dimensions } from 'react-native';
const { height } = Dimensions.get('window');


<View style={{ position: 'absolute' }}>
        <Swiper
          height={height}
          removeClippedSubviews={false}
        >
        ...
...
```

Hope it helps.

**Comment by qinfensky** [comment]
timer invalid? @gusgard

**Comment by gusgard** [comment]
@qinfensky I didn't use any `setTimeOut`, if that is what you are asking

**Comment by cielu** [comment]
@qinfensky  不需要设置100，设置0就好了

**Comment by sayhei** [comment]
@gusgard your solution solve my problems,thx

**Comment by zoofei** [comment]
@qinfensky it work! thank

**Comment by wangweifeng66** [comment]
After I added the timer, I still could not display the picture and needed to slide manually @qinfensky

**Comment by qinfensky** [comment]
@wangweifeng66 拼音？那就说中文啦，抱歉，我没在用这个组件了，现在具体细节我不知道了……

**Comment by montzkie18** [comment]
Hi guys, I ran into this issue as well on Android. Try setting animationEnabled to false on TabNavigatorConfig. That seems to fix the issue for me.

**Comment by d-gubert** [comment]
Disabling the animation works, but is a really big drawback for the usability of the TabNavigator component.

I'm looking for a better solution. Any help would be very appreciated :)

**Comment by sunnynegi** [comment]
Getting a same issue, Inside tabNavigator swiper content is not displaying, Any help provided on this issue.

**Comment by wutianlin** [comment]
Hi Guys,it looks like something wrong in showing images like this:
![image](https://user-images.githubusercontent.com/29848308/32640247-5826474e-c58d-11e7-9c83-182fd69d0d3b.png)
I have been add the timer but it still no work,I have no idea could you please help me?

**Comment by wutianlin** [comment]
Here are the timers:
 componentDidMount(){
        setTimeout(()=>{
            this.setState({
                swiperShow:true
            });
        },0)
    }

**Comment by Hancoson** [comment]
@qinfensky  不错，可以解决问题

**Comment by qinfensky** [comment]
> @qinfensky为什么它的剂量有效？根本原因是什么？

I hadn't used this component for a long time, and I was just trying to delay display and I was able to solve my problem. Now that "Swiper" and "React-Navigation" have been updated a lot, I guess the problem has been solved and I haven't had time to test it yet.
